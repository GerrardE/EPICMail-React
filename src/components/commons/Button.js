import React from 'react';
import PropTypes from 'prop-types';

export const ButtonLink = ({href, linkClass, linkName}) => {
  return (
    <a href={href} className={linkClass}>{linkName}</a>
  )
}

export const Button = ({btnClass, btnName, type, onClick, name}) => {
  return (
    <button type={type} className={btnClass} onClick={onClick} name={name}>{btnName}</button>
  )
}

ButtonLink.propTypes = {
  href: PropTypes.string,
  linkClass: PropTypes.string,
  linkName: PropTypes.string,
}

Button.propTypes = {
  btnClass: PropTypes.string,
  btnName: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  onClick: PropTypes.func.isRequired
}