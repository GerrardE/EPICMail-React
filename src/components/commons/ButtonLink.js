import React from 'react';
import PropTypes from 'prop-types';

export const ButtonLink = ({href, btnClass, btnName}) => {
  return (
    <a href={href} className={btnClass}>{btnName}</a>
  )
}

ButtonLink.propTypes = {
  href: PropTypes.string,
  btnClass: PropTypes.string,
  btnName: PropTypes.string,
}