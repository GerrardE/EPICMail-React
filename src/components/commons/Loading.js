import React from 'react';
import loader from './material-loader.gif';

export const Loading = () => {
  return (
    <div>
      <img src={loader} alt="loading..." className="loader"></img>
    </div>
  )
}
