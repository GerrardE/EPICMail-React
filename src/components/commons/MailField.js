import React from 'react';
import PropTypes from 'prop-types';
import { ButtonLink } from './Button';

export const MailField = ({ id, classType, email, createdon, title, body }) => {
  return (
    <div key={id} className={classType}>
      <div className="top">
        <span className="title">{email}</span>
        <span className="time-right">{createdon}</span>
      </div>
      <p className="lead">{title}</p>
      <p className="para">{body}</p>
      <ButtonLink
        linkClass="viewbtn"
        linkName="VIEW"
      />
      <ButtonLink
        linkClass="deletebtn"
        linkName="DELETE"
      />
    </div>
  )
}

MailField.propTypes = {
  id: PropTypes.string,
  classType: PropTypes.string,
  email: PropTypes.string,
  createdon: PropTypes.string,
  title: PropTypes.string,
  body: PropTypes.string,
  btnName: PropTypes.string,
}
