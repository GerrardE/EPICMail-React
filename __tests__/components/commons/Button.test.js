import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Button, ButtonLink } from '../../../src/components/commons/Button';

const props = {
  buttonLink: {
    href: '',
    linkClass: '',
    linkName: '',
    onClick: jest.fn()
  },
  button: {
    value: 1, 
    btnClass:'', 
    btnName: '', 
    type:'', 
    onClick: jest.fn(), 
    name: '',
  }
}

const setupButton = () => {
  const wrapper = shallow(<Button {...props.button} />);
  return wrapper
}

const setupButtonLink = () => {
  const wrapper = shallow(<ButtonLink {...props.buttonLink} />);
  return wrapper
}

describe('<Button/>', () => {
  test('should render the Button component properly', () => {
    let wrapperBtn;
    wrapperBtn = setupButton();
    expect(toJson(wrapperBtn)).toMatchSnapshot();
  })

  test('should render the ButtonLink component properly', () => {
    let wrapper;
    wrapper = setupButtonLink();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
