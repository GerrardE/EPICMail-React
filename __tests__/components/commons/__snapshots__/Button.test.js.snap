// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render the Button component properly 1`] = `
<button
  className=""
  name=""
  onClick={[MockFunction]}
  type=""
  value={1}
/>
`;

exports[`<Button/> should render the ButtonLink component properly 1`] = `
<a
  className=""
  href=""
  onClick={[MockFunction]}
/>
`;
